components/AllocatedWorkers/AddAllocatedWorker/AddAllocatedWorker.jsx

```diff
@@ -15,80 +15,62 @@ import {
 } from 'utils/api/allocatedWorkers';
 
 const AddAllocatedWorker = ({ personId }) => {
-  const [workersLoading, setWorkersLoading] = useState(false);
-  const [teams, setTeams] = useState();
-  const [workers, setWorkers] = useState();
-  const [error, setError] = useState();
   const [postError, setPostError] = useState();
   const [postLoading, setPostLoading] = useState();
-  const { push } = useRouter();
-  const { handleSubmit, register, watch } = useForm({
-    mode: 'onChange',
+  const { query, push, replace, pathname } = useRouter();
+  const { handleSubmit, register } = useForm({
+    defaultValues: query,
   });
   const { user } = useAuth();
-  const getAllocatedTeams = useCallback(async () => {
-    try {
-      const data = await getTeams();
-      setTeams(data.teams);
-    } catch (e) {
-      setError(true);
-    }
-  });
-  useEffect(() => {
-    getAllocatedTeams();
-  }, []);
-  const formValues = watch();
-  const getAllocatedWorkers = useCallback(async (selectedTeam) => {
-    setWorkersLoading(true);
-    try {
-      const data = await getTeamWorkers(selectedTeam);
-      setWorkers(data.workers);
-    } catch (e) {
-      setError(true);
-    }
-    setWorkersLoading();
-  });
-  useEffect(() => {
-    formValues.team && getAllocatedWorkers(formValues.team);
-  }, [formValues.team]);
-  useEffect(() => {
-    setPostError();
-  }, [formValues.team, formValues.worker]);
-  const addWorker = useCallback(async ({ worker }) => {
+  const { data: { teams } = {}, error: errorTeams } = getTeams();
```

Comment:
Never seen this syntax before. From the look of it, its getting all the teams, is ``data`` an object of teams?

Reply:
It means define `const teams` as `data.teams`.

`= {}` means that if data is `undefined` or `null` is not breaking, as it's falling back to `{}`. Otherwise, you will get `impossible to get teams of undefined`.